.github/workflows/build-and-test.yml

```diff
@@ -153,27 +146,19 @@ jobs:
           sudo apt-get update
           sudo apt-get install libudev-dev libusb-1.0-0-dev
 
-      - name: Cache cargo registry
-        uses: actions/cache@v3
-        with:
-          path: ~/.cargo/registry
-          # Add date to the cache to keep it up to date
-          key: ${{ matrix.os }}-stable-cargo-registry-${{ hashFiles('**/Cargo.lock') }}-${{ env.CURRENT_DATE }}
-          # Restore from outdated cache for speed
-          restore-keys: |
-            ${{ matrix.os }}-stable-cargo-registry-${{ hashFiles('**/Cargo.lock') }}
-            ${{ matrix.os }}-stable-cargo-registry-
-
-      - name: Cache cargo index
+      - name: Cache CARGO_HOME
         uses: actions/cache@v3
         with:
-          path: ~/.cargo/git
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
           # Add date to the cache to keep it up to date
-          key: ${{ matrix.os }}-stable-cargo-index-${{ hashFiles('**/Cargo.lock') }}-${{ env.CURRENT_DATE }}
+          key: ${{ matrix.os }}-stable-cargo-home-${{ env.CURRENT_DATE }}
```

Comment:
Do you think it's useful to have this date here? We had some doubts, but I'm not 100% sure what effect it has

Reply:
I think I was wrong in adding the `hashFiles('**/Cargo.lock')` because we don't use a lockfile for the crate. Since GitHub Actions only saves a new cache when there isn't a hit for the exact key ([see docs](https://github.com/actions/cache/blob/main/tips-and-workarounds.md#update-a-cache)), we need a key that changes. Usually the `Cargo.lock` would be good but we don't have one for the crate. If we keep using something like `${{ matrix.os }}-stable-cargo-home`, the cache will never be updated as there will always be a hit on the exact key. As an alternative though, we could also use something like week number (e.g. `2022-01` for Jan 1-7) if creating a new cache every day is excessive. The main issue is that not having a `Cargo.lock` for the crate means that caching `CARGO_HOME` becomes a bit more difficult

Reply:
I would say that current date is fine, as thats still a day. Usually we run a pr a few times per day, and then its merged or takes a while. so 1 week wouldnt really improve much